tests/shellcheck/test_shellcheck_gl_source.py

```diff
@@ -0,0 +1,116 @@
+from codecs import ignore_errors
+from curses import ERR
+from distutils.log import ERROR
+import logging
+import glob
+import os
+import pytest
+import re
+
+from pathlib import Path
+from subprocess import PIPE, run
+from itertools import filterfalse
+
+logger = logging.getLogger(__name__)
+
+SKIP_COMMENT = "skip-shellcheck"
+
+
+def is_bash_script(filepath):
+    try:
+        with open(filepath) as file:
+            head = file.readline()
+    except UnicodeDecodeError:
+        return False  # Ignore binary data
+    if 'bash' in head:
```

Comment:
I'm not quite sure if we never would have 'sh' within the shebang line?

maybe
```
if 'bash' or 'sh' in head:
```